IRIS FLOWER CLASSIFICATION

Using the measurements(sepal length,sepal width,petal length,petal width)as input feature, lets train a machine learning model to classify the iris flower into their species(setosa,versicolor,virginica).

In [ ]:
from google.colab import drive
drive.mount('/content/drive')
file='/content/drive/MyDrive/Iris.csv'




Mounted at /content/drive


Import all the libraries to be used

In [ ]:
#import your library
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score,confusion_matrix,classification_report,precision_score,recall_score,confusion_matrix,roc_auc_score,roc_curve,auc,f1_score



In [ ]:
#To load the file
df=pd.read_csv(file)
df.head

<bound method NDFrame.head of       Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
0      1            5.1           3.5            1.4           0.2   
1      2            4.9           3.0            1.4           0.2   
2      3            4.7           3.2            1.3           0.2   
3      4            4.6           3.1            1.5           0.2   
4      5            5.0           3.6            1.4           0.2   
..   ...            ...           ...            ...           ...   
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
0       Iris-setosa  
1       Iris-setosa  
2       Iris-setosa  
3       Iris-setosa  
4       Iris-setosa  
..              ...  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica  

[150 rows x 6 columns]>

In [ ]:
#view the first five column
df.head()


Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [ ]:
#splitting the dataset
x=df.drop(['Id','Species'],axis=1)
y=df['Species']
print(x)
print(y)

     SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
0              5.1           3.5            1.4           0.2
1              4.9           3.0            1.4           0.2
2              4.7           3.2            1.3           0.2
3              4.6           3.1            1.5           0.2
4              5.0           3.6            1.4           0.2
..             ...           ...            ...           ...
145            6.7           3.0            5.2           2.3
146            6.3           2.5            5.0           1.9
147            6.5           3.0            5.2           2.0
148            6.2           3.4            5.4           2.3
149            5.9           3.0            5.1           1.8

[150 rows x 4 columns]
0         Iris-setosa
1         Iris-setosa
2         Iris-setosa
3         Iris-setosa
4         Iris-setosa
            ...      
145    Iris-virginica
146    Iris-virginica
147    Iris-virginica
148    Iris-virginica
149    Iris-

In [ ]:
#shape of x and y
x.shape,y.shape


((150, 4), (150,))

Train the dataset

In [ ]:
#using thetrain_test_split function to split data
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2,random_state=42)

In [ ]:
#test the equality of xtrain and ytrain
x_train.shape[0]==y_train.shape[0]



True

In [ ]:
#Building our model using logistic regression
#creating an instance of a model
logreg=LogisticRegression(max_iter=200,multi_class='ovr')
logreg

LogisticRegression(max_iter=200, multi_class='ovr')

In [ ]:
log_model=LogisticRegression(max_iter=200,multi_class='ovr')
log_model

LogisticRegression(max_iter=200, multi_class='ovr')

In [ ]:
#fitting our model
#Train set is for training
#test set is for evaluation
log_model.fit(x_train,y_train)


/usr/local/lib/python3.12/dist-packages/sklearn/linear_model/_logistic.py:1256: FutureWarning: 'multi_class' was deprecated in version 1.5 and will be removed in 1.7. Use OneVsRestClassifier(LogisticRegression(..)) instead. Leave it to its default value to avoid this warning.
  warnings.warn(


LogisticRegression(max_iter=200, multi_class='ovr')

In [ ]:

#model evaluation
#making predictions
log_pred=log_model.predict(x_test)
log_pred


array(['Iris-versicolor', 'Iris-setosa', 'Iris-virginica',
       'Iris-versicolor', 'Iris-versicolor', 'Iris-setosa',
       'Iris-versicolor', 'Iris-virginica', 'Iris-versicolor',
       'Iris-versicolor', 'Iris-virginica', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-virginica', 'Iris-virginica',
       'Iris-versicolor', 'Iris-versicolor', 'Iris-virginica',
       'Iris-setosa', 'Iris-virginica', 'Iris-setosa', 'Iris-virginica',
       'Iris-virginica', 'Iris-virginica', 'Iris-virginica',
       'Iris-virginica', 'Iris-setosa', 'Iris-setosa'], dtype=object)

In [ ]:
#to find the accuracy
accuracy_score(y_test,log_pred)

0.9666666666666667

In [ ]:
#f1 score
f1_score(y_test,log_pred,average='weighted')

0.9664109121909632

In [ ]:
#precision score
precision_score(y_test,log_pred,average='weighted')

0.9694444444444444

In [ ]:
#plot confusion matrix
cm=confusion_matrix(y_test,log_pred)
cm




array([[10,  0,  0],
       [ 0,  8,  1],
       [ 0,  0, 11]])

from the above array, its a 3x3 matrix meaning it contains the 3 classification of the species of iris flower(sertosa,versicolor and virginica

MODEL PERFORMANCE

The model achieved a very high performance of about 0.966=97%, Setosa was classified well,Virginica was also classified  well and then for versicolor there was a slight misclassification with virginica

Precision and F1 score were 0.96, greater than 0.95 confirming that the model generalizes well.

KEY INSIGHTS

The measurements Petal width,petal length,sepal width,sepal length give enough seperation to distinguish betwen them, Setosa is linearly seperable  while Versicolor and virginica overlap slightly which explains the misclassification

CONCLUSION

The trained machine model is highly effective,classifying the iris flower species with the various measurement,It demonstrates that any number of measurement can be modeled accuratel with machine learning technique

Setosa is easiest to classify while versicolor and virginico is slightly harder but still has a good accuracy.This makes

In [ ]:
#Question 2.

UNEMPLOYMENT ANALYSIS WITH INDIA

In [ ]:
from google.colab import drive
drive.mount('/content/drive')
file='content/MyDrive/drive/Unemploymentin India.csv'



Lets import the libraries to be used

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as pt
import seaborn as sns
stylerr='ggplot'
from matplotlib import style

#Importing SKlearn functions and classes
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import OneHotEncoder
from sklearn.metrics import accuracy_score,confusion_matrix,classification_report,precision_score,recall_score,confusion_matrix,roc_auc_score,roc_curve,auc,f1_score



Exploring the dataset

In [ ]:
df=pd.read_csv(file)
df.head


